cmd/localapiservice/localapishim.go

```diff
@@ -198,8 +198,8 @@ func Java_com_tailscale_ipn_ui_notifier_Notifier_startIPNBusWatcher(
 			return true
 		}
 		jni.Do(shim.jvm, func(env *jni.Env) error {
-			jjson := jni.JavaString(env, string(js))
-			onNotify := jni.GetMethodID(env, shim.notifierClass, "onNotify", "(Ljava/lang/String;Ljava/lang/String;)V")
+			jjson := jni.NewByteArray(jenv, js)
```

Comment:
Passing ByteArray instead of a String seemed to help quite a bit with memory usage.